packages/styleguide/src/components/Chart/Table.js

```diff
@@ -254,18 +284,107 @@ Table.propTypes = propTypes
 export default Table
 
 const Cell = (props) => {
-  const { type, width, color, colorScale, value, children, isNumeric } = props
+  const {
+    columnName,
+    width,
+    color,
+    colorScale,
+    value,
+    children,
+    isNumeric,
+    isBarColumn,
+    barChartExtent,
+    numberFormatter,
+  } = props
+
+  // const [parentWidth, setParentWidth] = useState(null)
+  // const parentCell = useCallback((node) => {
+  //   if (node !== null) {
+  //     setParentWidth(node.getBoundingClientRect().width)
+  //   }
+  // })
+
+  const parentWidth = Math.max(width, 100)
+
+  const barScale = scaleLinear()
+    .domain(barChartExtent)
+    .range([5, parentWidth - PADDING])
+
   return (
     <td
+      // ref={parentCell}
       {...(isNumeric && styles.cellNumeric)}
       {...styles.cell}
       style={{
         width: width !== undefined ? +width : undefined,
-        backgroundColor: color ? colorScale(value) : 'transparent',
-        color: color && getTextColor(colorScale(value)),
+        backgroundColor:
+          !isBarColumn && color ? colorScale(value) : 'transparent',
+        color: !isBarColumn && color && getTextColor(colorScale(value)),
+        whiteSpace: isBarColumn && 'nowrap',
       }}
     >
-      {children}
+      {isBarColumn ? (
+        <BarComponent
+          parentWidth={parentWidth - PADDING}
+          colorScale={colorScale}
+          barWidth={barScale(value)}
+          label={numberFormatter(value)}
+          columnName={columnName}
+          color={color}
+        />
+      ) : (
+        children
+      )}
     </td>
   )
 }
+
+const BarComponent = (props) => {
+  const { parentWidth, barWidth, color, label, columnName, colorScale } = props
+
+  const labelSize = labelGauger(label)
+  const BAR_LABEL_PADDING = 10
+
+  const isLabelOutside = labelSize > +barWidth - BAR_LABEL_PADDING
```

Comment:
Note sure if the `BAR_LABEL_PADDING` is needed, and `barWidth` comes from scale – doesn't need casting afais
```suggestion
  const isLabelOutside = labelSize > barWidth
```